### ISSUE TITLE
Sharing an Image on Facebook no longer works

### ISSUE THREAD

**Issue by tanis2000** [ideashower/ShareKit#72]
I have verified this both within my application and the example project included in ShareKit. 
If you share an image on Facebook, you get authorized but then it says that the image has been succesfully shared while the truth is that nothing has been shared. 

If there's anything else I can provide to help you debug this issue, just let me know.

**Comment by ideashower** [comment]
Can you check your Facebook albums?  I've heard of cases where your photos will get pushed into a place where you have to 'approve' them.

**Comment by tanis2000** [comment]
You are right. They get pushed to an album with the name of the application (which also happened before btw) but they do not get shared on your profile with a description as they used to till a couple of days ago. 

So I guess that the guys at FB must have changed something. Are you sure they haven't changed the API so that the one used by ShareKit is only used to store photos in an album while you also have to call some other function to post that very same photo on your wall to share with friends?

**Comment by traustitj** [comment]
Are there any news on this ?  Or is there another way to share images on your wall ?

**Comment by romulusx** [comment]
Since it looks like Facebook is pushing their latest Connect code: ( http://github.com/facebook/facebook-ios-sdk ) and modifying the behavior of their old API, it may be cleaner to run with issue 69 (support Graph API). Either way, to support Wall image posts with useful textual info through the new api, or old REST publish_stream, images need to be provided via links (the image needs to be hosted elsewhere, not Facebook), before Facebook caches them.

**Comment by tanis2000** [comment]
I wonder if it could be done in two steps:

a) post the photo to FB (the pic gets uploaded to an album)
b) publish the photo that is now in your album to your wall

**Comment by romulusx** [comment]
I'm not positive, but I don't think that will work since images in albums are url located in facebook's cdn (fbcdn.net), and that is where facebook explicitly says do not link to when posting to the feed/wall. 

I've had to use the REST api ( just noticed the Graph api does not yet support action links as of today) and an offsite storage area (currently using Amazon S3, which is very inexpensive) for including images in wall posts. Works great.

**Comment by liphoto** [comment]
this was giving me the most problems while trying to first implement sharekit today. tried different keys and secrets and nothing worked. 
Turns out its this exact problem with images, went to check my albums and blank images were there and the folder was named my as my app from facebook. no sharing was working on the wall at all.
